src/Provider/EmailCodesProvider.php

```diff
@@ -10,25 +10,55 @@
 
 namespace BitBag\SyliusMailTemplatePlugin\Provider;
 
+use BitBag\SyliusMailTemplatePlugin\Repository\EmailTemplateRepositoryInterface;
+
 final class EmailCodesProvider implements EmailCodesProviderInterface
 {
     private array $emails;
 
-    public function __construct(array $emails)
+    private EmailTemplateRepositoryInterface $emailTemplateRepository;
+
+    public function __construct(array $emails, EmailTemplateRepositoryInterface $emailTemplateRepository)
     {
         $this->emails = $emails;
+        $this->emailTemplateRepository = $emailTemplateRepository;
     }
 
     public function provideWithLabels(): array
+    {
+        return $this->getTypesWithLabels();
```

Comment:
Can you remove private method by putting its logic here and use the public one method everywhere the private was used?

Reply:
Done